File: frontend/src/dto/order.dto.ts
import { OrderLineDTO } from "./orderLine.dto";

export interface OrderDTO {
    orderId: string;
    tableTag: string;
    areaName: string;
    status: OrderStatus;
    totalPrice: number;
    orderLines: OrderLineDTO[];
    createdAt: string;
    updatedAt: string;
};

export enum OrderStatus {
    EATING = "EATING",
    COMPLETED = "COMPLETED",
    CANCELLED = "CANCELLED"
};

export interface UpdateOrderStatusRequest {
    orderId: string;
    orderStatus: OrderStatus;
}

export interface UpdateOrderStatusResponse {
    successful: boolean;
    previousStatus: OrderStatus;
    newStatus: OrderStatus;
}

File: backend/src/main/java/com/example/backend/dto/OrderDTO.java
package com.example.backend.dto;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

import com.example.backend.entities.OrderStatus;

public class OrderDTO {
    private UUID orderId;
    private String tableTag;
    private String areaName;
    private OrderStatus status;
    private BigDecimal totalPrice;
    private List<OrderLineDTO> orderLines;
    private String createdAt;
    private String updatedAt;

    public UUID getOrderId() {
        return orderId;
    }

    public void setOrderId(UUID orderId) {
        this.orderId = orderId;
    }

    public String getTableTag() {
        return tableTag;
    }

    public void setTableTag(String tableTag) {
        this.tableTag = tableTag;
    }

    public BigDecimal getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(BigDecimal totalPrice) {
        this.totalPrice = totalPrice;
    }

    public List<OrderLineDTO> getOrderLines() {
        return orderLines;
    }

    public void setOrderLines(List<OrderLineDTO> orderLines) {
        this.orderLines = orderLines;
    }

    public OrderStatus getStatus() {
        return status;
    }

    public void setStatus(OrderStatus status) {
        this.status = status;
    }

    public String getAreaName() {
        return areaName;
    }

    public void setAreaName(String areaName) {
        this.areaName = areaName;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(String updatedAt) {
        this.updatedAt = updatedAt;
    }
    
}
